### PR TITLE
Descend into newly created folder by clicking enter again

### LINKED ISSUE #1905: New folder creation flow: descend into newly created folder by clicking enter again
When finished with entering the new folder name, the most likely thing people can do with it is fill it up with files. That’s why we should automatically descend into it.

With #1618 (share current folder) it will also be possible to share it from inside so that’s taken care of too.

cc @nextcloud/javascript @nextcloud/designers

### PR DESCRIPTION
**Pull request by sagorika1996**
Fixes #1905

### CODE REVIEW

**Comment by MorrisJobke** [comment]
cc @nextcloud/javascript

File: apps/files/js/filelist.js [hunk 1]
```diff
@@ -1433,6 +1433,12 @@
 				$tr.addClass('appear transparent');
 				window.setTimeout(function() {
 					$tr.removeClass('transparent');
+					$("#fileList tr").each(function(index) {
```
Comment (juliushaertl) [inline apps/files/js/filelist.js]: Just using `$('#fileList tr').removeClass('mouseOver');` should be enough here.

**Review by juliusknorr — CHANGES_REQUESTED** [review]
You can make the code a bit simpler, see the inline comments. Seems a bit of a hacky solution to me, but works. :wink:

File: apps/files/js/filelist.js [hunk 2]
```diff
@@ -1433,6 +1433,12 @@
...
+						if ($(this).hasClass("mouseOver")) {
```
Comment (ChristophWurst) [inline apps/files/js/filelist.js]: I guess it's fine to just call `.removeClass`

**Review by juliusknorr — APPROVED** [review]

**Comment by MorrisJobke** [comment]
CI failure is unrelated.

**Comment by MorrisJobke** [comment]
For me this does nothing except adding CSS classes obviously. Tested in Safari.

**Comment by sagorika1996** [comment]
awesomescreenshot-2018-03-11t15-44-23-853z

@MorrisJobke That's odd, its working fine in Chrome.

**Comment by sagorika1996** [comment]
@MorrisJobke @juliushaertl @skjnldsv ping

**Comment by skjnldsv** [comment]
Looking at the code, I'm with @MorrisJobke, I don't see how that could work. What triggers the change?
Do we have an auto-focus on the folder's name link?

**Comment by sagorika1996** [comment]
When enter key is pressed, click action is triggered on the file which has the mouseOver class
https://github.com/nextcloud/server/blob/64feb3f3912f0976bb028f5d56a6434c55085925/apps/files/js/keyboardshortcuts.js#L101-L108

**Review by MorrisJobke — APPROVED** [review]
Tested and works 🙈

**Pull request merged**